ios/Classes/PushNotificationEventHandlers.swift

```diff
@@ -59,9 +59,33 @@ public class PushNotificationEventHandlers: NSObject, UNUserNotificationCenterDe
         if (application.applicationState == .background || application.applicationState == .inactive) {
             methodName = AblyPlatformMethod_pushOnBackgroundMessage
         }
-        let remoteMessage = RemoteMessage(data: userInfo._bridgeToObjectiveC(), notification: nil);
+        
+        let notification: Notification? = createNotification(from: userInfo);
+        let remoteMessage = RemoteMessage(data: userInfo._bridgeToObjectiveC(), notification: notification);
         methodChannel.invokeMethod(methodName, arguments: remoteMessage) { flutterResult in
             completionHandler(.newData);
         }
     }
+    
+    private func createNotification(from userInfo: [AnyHashable : Any]) -> Notification? {
+        let notification = Notification();
+        
+        if let aps = userInfo["aps"] as? NSDictionary {
+            if let alert = aps["alert"] as? NSDictionary {
+                if let title = alert["title"] as? String {
+                    notification.title = title
+                }
+                if let body = alert["body"] as? String {
+                    notification.body = body
+                }
+            }
+        }
+        
+        if (notification.title != nil || notification.body == nil) {
```

Comment:
You are returning an object if both values are nil, this is what you want?

Reply:
This logic has actually been removed since I am now I am creating a notification regardless of the content. However I think you misunderstand the code there. I am returning the notification if the title OR body is not nil (there is _some_ data).

Reply:
for me: 
if title is nil OR body is nil (==) -> you return notification object
if title is NOT nil -> you return the notification object

if you want to have "some data" the second condition should look like: `notification.body != nil`, it's equal to `title OR body` is not nil 

This is how I understand this

Reply:
Ohhhh. I did not notice the `==`, which should actually be `!=`. Thankfully, this code is removed 😂